sink/sink

```diff
@@ -303,6 +303,31 @@ class Buffer(object):
         result += os.read(self.fd, 1024)
         return result
 
+# Create the directory in a race free way
+def mkdir_and_chdir(base, identifier):
+    directory = os.path.abspath(os.path.join(base, identifier))
+
+    # 1. Create a temporary non-empty directory and get its handle
+    tempdir = tempfile.mkdtemp(prefix=identifier, dir=base)
+    os.close(os.open(os.path.join(tempdir, ".sink"), os.O_WRONLY | os.O_CREAT))
```

Comment:
Renaming over an empty directory does not result in EEXIST. Therefore in order to claim this directory (and not have someone blow it away) it needs an dir entry in it.